smoke.py

```diff
@@ -87,6 +91,15 @@ def __del__(self):
 class BaseRuntimeSmokeTest(Test):
 
     def setUp(self):
+        compdir = self.params.get('compiler-test-dir')
+        if compdir is None:
+            self.error("Compiler test resource directory must be supplied")
```

Comment:
Why does this need to be supplied explicitly? Is there any reason to believe we couldn't just assume its location is fixed (relative to the smoke.py script)?

I probably should have caught that for the mockcfg patches earlier as well.